lib/cErrors.mli

```diff
@@ -51,6 +57,15 @@ val user_err : ?loc:Loc.t -> ?info:Exninfo.info -> Pp.t -> 'a
     handled by registered handlers.
 *)
 
+module type Register = sig
+  type e
+  type _ tag += T : e tag
+  val pp : e -> Pp.t
```

Comment:
I guess one could ask a string containing the rst which is then printed in the manual and the rst label as well. @jfehrle what is unclear to me is how to generate a link to the doc as part of the message.

Asking coq to spit att errors it knows is possible, assuming you load all plugins being distributed with it. The patch was, and still is, easy.

Reply:
What's the patch without this PR?

Reply:
or even with it considering all the user_err we have

Reply:
> what is unclear to me is how to generate a link to the doc as part of the message

Meaning a clickable hyperlink in the GUI that opens up the documentation at the relevant point?  It could be done, but maybe not worth the trouble--isn't the current index of messages easy enough to use?

For printing the list of all messages, something like this would be convenient for getting them in the documentation:
- Messages are represented as a single Printf string.
- Insertions are identified with `%s`.  Extra credit if the list tells you the nonterminal name associated with the message, e.g. `@ident` or `@term.  Double extra credit if the `@ident` is in the string and OCaml typing checks that the type is correct.
- Add tooling to `doc_grammar` to match messages in the doc to the dumped list.  Automatic update is probably not feasible.
- Try to avoid building wording variations into the messages (e.g. singular vs plural variations)

Reply:
> a clickable hyperlink in the GUI that opens up the documentation

It does not have to be limited to a GUI; it could be used for terminal output too:
```ocaml
Printf.printf "\x1b]8;;%s\x1b\\%s\x1b]8;;\x1b\\" "http://coq.inria.fr/" "Coq's main page";;
```

Reply:
> Asking coq to spit all errors it knows is possible, assuming you load all plugins being distributed with it. The patch was, and still is, easy.

I meant that patching coq so that it does so is easy (this PR does not change this), IMO.

Reply:
>I meant that patching coq so that it does so is easy (this PR does not change this), IMO.

I don't think it's easy.